src/utils/CachedItem.ts

```diff
@@ -0,0 +1,26 @@
+class CachedItem<T> {
+  private cache: T | null;
+
+  private ttl: number;
+
+  constructor(ttl: number) {
+    this.cache = null;
+    this.ttl = ttl;
+  }
+
+  get(): T | null {
+    // retrieves an item from the cache or returns null if no item is found
+    return this.cache;
+  }
+
+  set(value: T): void {
+    // adds an item to the cache
+    this.cache = value;
+    setTimeout(() => {
+      this.cache = null;
+    }, this.ttl);
```

Comment:
What alternatives did you consider here? Setting timeouts is a little heavy-handed and harder to trace.

Reply:
I also believe this will lead to a bug if the user runs `set` several times within too small of a time interval. 

Suppose we have the following:

```ts
const cachedItem = new CachedItem<number>(500); // 500 ms ttl
cachedItem.set(123); 
setTimeout(() => cachedItem.set(456), 250);
```
If we try to extract the cached item 250ms after it was set the second time, we'll get null.